interface/resources/qml/hifi/avatarapp/AvatarsModel.qml

```diff
@@ -21,17 +21,47 @@ ListModel {
 
         return marketItemUrl;
     }
+	
+    function makeMarketThumbnailUrl(marketId) {
+        var avatarThumbnailUrl = "https://hifi-metaverse.s3-us-west-1.amazonaws.com/marketplace/previews/%marketId%/large/hifi-mp-%marketId%.jpg"
+            .split('%marketId%').join(marketId);
+            
+        return avatarThumbnailUrl;
+    }
+	
+    function trimFileExtension(url) {
+        var trimmedUrl = url.substring(0, (url.indexOf("#") == -1) ? url.length : url.indexOf("#"));
+        trimmedUrl = trimmedUrl.substring(0, (trimmedUrl.indexOf("?") == -1) ? trimmedUrl.length : trimmedUrl.indexOf("?"));
```

Comment:
use strict comparisons change `==` to `===` which strictly checks it for the type as well.

Reply:
I mean, is this a recognizable coding pattern in JS?

Like you could this in any other language, but you almost always would do it differently: use interpolation, a replace() call, sprintf, that kind of thing.

So the question is whether split and replace used for this particular purpose is a common feature of JS programs, or if most JS programmers would take a different approach here.

Reply:
There has been constant back and forth about the efficiency of replace vs split/join in JS over the past 10 years. In modern browsers they say that replace has been made more efficient so it might very well be the case with QT, when this file was originally made, that split/join was more efficient or still is. I was under the impression that that line of code had passed hifi's code review and that I should not fix what is not broken.